ballerina/vector-store.bal

```diff
@@ -89,17 +89,55 @@ public distinct isolated class InMemoryVectorStore {
     # + return - An array of vector matches sorted by similarity score (limited to topK), 
     # or an `ai:Error` if the query fails
     public isolated function query(VectorStoreQuery query) returns VectorMatch[]|Error {
-        if query.embedding !is Vector {
-            return error Error("InMemoryVectorStore supports dense vectors exclusively");
-        }
-
         lock {
-            VectorMatch[] sorted = from var entry in self.entries
+            VectorStoreQuery clonedQuery = query.cloneReadOnly();
+            Embedding? embedding = clonedQuery.embedding;
+            MetadataFilters? filters = clonedQuery.filters;
+            if embedding !is Vector? {
+                return error Error("InMemoryVectorStore supports dense vectors exclusively");
+            }
+            if embedding is () {
+                readonly & VectorMatch[] results;
+                if filters is () {
+                    results = from InMemoryVectorEntry entry in self.entries
+                    select {
+                        chunk: entry.chunk.cloneReadOnly(),
+                        embedding: entry.embedding.cloneReadOnly(),
+                        similarityScore: 1.0,
```

Comment:
should we change the similarity score in these case 0? since we don't have a target embedding we can't calculate a similarity score. According to our documentation the similartiyScore here is the similarity between vectors not metadata. WDYT?